### PR TITLE
Remove the Firefox extension building code.

### PR DESCRIPTION
**Pull request by brendandahl**
Firefox no longer supports this legacy extension and it is a pain to
sync changes from here and mozilla central.

There's still a bit more cleanup we could do for locale's, but I'd like to leave that for a follow up bug.

### CODE REVIEW

**Comment by timvandermeij** [comment]
I think we also need to update both bots so that links to the extension are no longer in the previews.

Firefox still maintains its own version of the extension upstream, which is embedded in each Firefox release, right?

**Review by Snuffleupagus — COMMENTED** [review]
While I completely understand, and agree with, the reasoning behind this PR there's one important caveat to consider: Note how we're currently handling preferences in the extension *and*, most importantly, the mozilla-central code.

Currently we're simply bundling https://github.com/mozilla/pdf.js/blob/master/web/default_preferences.json into the relevant files using the preprocessor; see https://github.com/mozilla/pdf.js/pull/9566/files#diff-fac62551eadfe79f141900988b8e8a8dL80 and https://github.com/mozilla/pdf.js/pull/9566/files#diff-c64446a1f7e4337255356cd285b15ca5L33.

This (currently) means that if any changes are made to the `default_preferences.json` file, those are *automatically* mirrored to the relevant mozilla-central files.

If we completely remove this code from the GitHub repo it would have the effect of making any preference changes a two-step process, where you'd need to *also* submit a mozilla-central patch simultaneously for the preference changes.

*Edit:* Please see PR #9571 for *one* possible way of addressing these issues.

**Comment by brendandahl** [comment]
> Firefox still maintains its own version of the extension upstream, which is embedded in each Firefox release, right? (It will probably become a WebExtension anyway, and we also have an open issue for that here too.)

Yeah.

**Comment by brendandahl** [comment]
@Snuffleupagus good to go now?

**Comment by Snuffleupagus** [comment]
> @Snuffleupagus good to go now?

We probably need to do one more mozilla-central update *first*, to sync the changes from PR #9571, but after that we should be good here :-)

File: README.md [hunk 1]
```diff
@@ -27,15 +27,9 @@ Feel free to stop by #pdfjs on irc.mozilla.org for questions or guidance.
...
+#### Firefox
 
-PDF.js is built into version 19+ of Firefox, however, one extension is still available:
-
-+ [Development Version](http://mozilla.github.io/pdf.js/extensions/firefox/pdf.js.xpi) - This extension is mainly intended for developers/testers, and it is updated every time new code is merged into the PDF.js codebase. It should be quite stable but might break from time to time.
-
-  + Please note that the extension is *not* guaranteed to be compatible with Firefox versions that are *older* than the current Nightly version, see the [Release Calendar](https://wiki.mozilla.org/RapidRelease/Calendar#Past_branch_dates).
-
-  + The extension *may* also work in Seamonkey, provided that it is based on a Firefox version as above (see [Which version of Firefox does SeaMonkey 2.x correspond with?](https://wiki.mozilla.org/SeaMonkey/FAQ#General)), but we do *not* guarantee compatibility.
+PDF.js is built into version 19+ of Firefox
```
Comment (Rob--W) [inline README.md]: Add a period at the end.

File: gulpfile.js [hunk 2]
```diff
@@ -862,43 +724,16 @@ gulp.task('mozcentral-pre', ['buildnumber', 'locale'], function () {
...
-        .pipe(replace(/\bPDFJSSCRIPT_PREF_PREFIX\b/g, MOZCENTRAL_PREF_PREFIX))
-        .pipe(gulp.dest(MOZCENTRAL_CONTENT_DIR)),
-    preprocessJS(FIREFOX_CONTENT_DIR + 'PdfJsNetwork.jsm', defines, true)
-        .pipe(gulp.dest(MOZCENTRAL_CONTENT_DIR)),
-    preprocessJS(FIREFOX_CONTENT_DIR + 'PdfjsContentUtils.jsm', defines, true)
-        .pipe(gulp.dest(MOZCENTRAL_CONTENT_DIR)),
-    preprocessJS(FIREFOX_CONTENT_DIR + 'PdfjsChromeUtils.jsm', defines, true)
-        .pipe(replace(/\bPDFJSSCRIPT_PREF_PREFIX\b/g, MOZCENTRAL_PREF_PREFIX))
-        .pipe(gulp.dest(MOZCENTRAL_CONTENT_DIR)),
+        .pipe(gulp.dest(MOZCENTRAL_CONTENT_DIR))
```
Comment (Rob--W) [inline gulpfile.js]: Add a comma at the end.

**Review by Rob--W — COMMENTED** [review]
> (It will probably become a WebExtension anyway, and we also have an open issue for that here too.)

That is #8533.

**Review by yurydelendik — APPROVED** [review]
LGTM: I think we need to remove that from `gulp web` (https://github.com/brendandahl/pdf.js/blob/1c3986a8e21ef8d1204fe0d8f465a4a51f03599c/gulpfile.js#L1071) and remove extension signing from botio

**Review by timvandermeij — APPROVED** [review]
Looks good to me.

**Comment by brendandahl** [comment]
/botio-linux preview
(test botio changes)

**Pull request merged**